cloudinit/config/cc_ca_certs.py

```diff
@@ -198,26 +246,37 @@ def handle(
     # If there is a remove_defaults option set to true, remove the system
     # default trusted CA certs first.
     if "remove-defaults" in ca_cert_cfg:
-        log.warning(
+        LOG.warning(
             "DEPRECATION: key 'ca-certs.remove-defaults' is now deprecated."
             " Use 'ca_certs.remove_defaults' instead."
         )
-        if ca_cert_cfg.get("remove-defaults", False):
-            log.debug("Removing default certificates")
-            remove_default_ca_certs(cloud.distro.name, distro_cfg)
+        if cloud.distro.name in ["fedora", "rhel"]:
+            LOG.error(
+                "Disabling default certificates is not supported for"
+                " this distro."
+            )
+        elif ca_cert_cfg.get("remove-defaults", False):
+            LOG.debug("Disabling default certificates")
+            disable_default_ca_certs(cloud.distro.name, distro_cfg)
     elif ca_cert_cfg.get("remove_defaults", False):
-        log.debug("Removing default certificates")
-        remove_default_ca_certs(cloud.distro.name, distro_cfg)
+        if cloud.distro.name in ["fedora", "rhel"]:
```

Comment:
This whole `if cloud.distro.name in ["fedora", "rhel"]:` and `else` could be put in a separate function instead of duplicating the code, it seems exactly the same to me.

Reply:
That code is gone now anyway as part of my reverting to old behaviour.